n3fit/src/evolven3fit_new/eko_utils.py

```diff
@@ -34,20 +43,45 @@ def construct_eko_cards(
     theory = Loader().check_theoryID(theoryID).get_description()
     theory.pop("FNS")
     theory.update(theory_card_dict)
-    theory_card = gen_theory.gen_theory_card(theory["PTO"], theory["Q0"], update=theory)
+    if "nfref" not in theory:
+        theory["nfref"] = 5
+    if "nf0" not in theory:
+        theory["nf0"] = 4
+    #The Legacy function is able to construct a theory card for eko starting from an NNPDF theory
+    legacy_class = runcards.Legacy(theory, {})
+    theory_card = legacy_class.new_theory
     # construct operator card
     q2_grid = utils.generate_q2grid(
         theory["Q0"],
         q_fin,
         q_points,
         {theory["mb"]: theory["kbThr"], theory["mt"]: theory["ktThr"]},
     )
-    op_card = gen_op.gen_op_card(q2_grid, update={"interpolation_xgrid": x_grid})
-    op_card.update(op_card_dict)
+    op_card = default_op_card
+    op_card.update(
+        {
+            "mu0": theory["Q0"],
+            "_mugrid": np.sqrt(q2_grid).tolist(),
+        }
+    )
+    op_card["rotations"]["xgrid"] = x_grid
+    # Specific defaults for evolven3fit evolution
+    op_card["configs"].update(EVOLVEN3FIT_CONFIGS_DEFAULTS)
+    # User can still change the configs via op_card_dict
+    
+    # Note that every entry that is not a dictionary should not be 
+    # touched by the user and indeed an user cannot touch them            
+    for key in op_card:
+        if key in op_card_dict and isinstance(op_card[key], dict):
+            op_card[key].update(op_card_dict[key])
+        elif key in op_card_dict:
+            _logger.warning("Entry %s is not a dictionary and will be ignored", key)
```

Comment:
I'm wondering if at this point it should just raise an error. If a code runs and returns something that looks halfway alright I don't check the logs. 

Whenever there is a key in a dictionary there may have been a reason for it to be there and if it was put in the wrong place/mistyped or something like that it's good if it raises an error. Probably outweights the few times it would raise an error when unnecessary.

Reply:
In general I agree. The point is that in this case the only thing that an user can pass and that would be ignored are the Q2 and x grids which, in the particular case of the postfit evolution, I don't thing will be source of confusion.

Reply:
What is the use case of `op_card_dict`? Is this something that the user is expected to use or is it more an "advance use case" / for debugging / etc?

Reply:
I would say that in general the user is not going to use it. The only real use case is if he wants to change the solution of the evolution or something like that and this is actually possible. The only two things that at the moment cannot be changed are the q-grid and the x-grid but I would say that they are usually not changed when one does the postfit evolution.